### PR TITLE
1099/sender override failed message logger

### PR DESCRIPTION
**Pull request by gudgl**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: consumers/sender/main.go [hunk 1]
```diff
@@ -45,6 +45,40 @@ type MessageHandler struct {
...
+		return
+	}
+
+	err = h.storage.CreateSendLog(&entities.SendLog{
+		ID:           msg.ID,
+		UserID:       msg.UserID,
+		CampaignID:   msg.CampaignID,
+		SubscriberID: msg.SubscriberID,
+		Status:       entities.SendLogStatusFailed,
+		Description:  "Failed to send email five times",
```
Comment (FilipNikolovski) [inline consumers/sender/main.go:70]:
```suggestion
		Description:  "Exceeded max attempts for sending the e-mail.",
```

File: consumers/sender/main.go [hunk 2]
```diff
@@ -45,6 +45,40 @@ type MessageHandler struct {
...
+	}
+
+	msg := new(entities.SenderTopicParams)
+	err := json.Unmarshal(m.Body, msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return
+	}
+
```
Comment (FilipNikolovski) [inline consumers/sender/main.go:63]: We should also add an error log here as well.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: consumers/sender/main.go [hunk 3]
```diff
@@ -61,13 +61,20 @@ func (h *MessageHandler) LogFailedMessage(m *nsq.Message) {
 		return
 	}
 
+	logrus.WithFields(logrus.Fields{
+		"send_log_id":   msg.ID.String(),
```
Comment (FilipNikolovski) [inline consumers/sender/main.go:65]: You don't need to invoke String() explicitly I think.

```suggestion
		"send_log_id":   msg.ID,
```

**Comment by Dzalevski** [comment]
@gudgl pull from master and clear go deps with `go mod tidy`

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**